src/3p.js

```diff
@@ -123,3 +123,51 @@ export function validateSrcContains(string, src) {
     throw new Error('Invalid src ' + src);
   }
 }
+
+/**
+ * Throws a non-interrupting exception if data contains a field not supported
+ * by this embed type.
+ * @param {!Object} data
+ * @param {!Array<string>} allowedFields
+ */
+export function checkData(data, allowedFields) {
+  // Throw in a timeout, because we do not want to interrupt execution,
+  // because that would make each removal an instant backward incompatible
+  // change.
+  try {
+    validateData(data, allowedFields);
+  } catch (e) {
+    setTimeout(() => {
+      throw e;
+    });
+  }
+}
+
+/**
+ * Throws an exception if data contains a field not supported
+ * by this embed type.
+ * @param {!Object} data
+ * @param {!Array<string>} allowedFields
+ */
+export function validateData(data, allowedFields) {
+  const defaults = {
```

Comment:
Do you feel strongly?

I renamed it to make the purpose more straight forward, but I did not move it.

- in practice this function is only called once per compilation unit (so there is no double allocation)
- it is harder to read when the constant is far away.